docs/examples/using-terraform-plugin.md

```diff
@@ -0,0 +1,265 @@
+---
+layout: default
+title: Spawn a virtual machine on each Cloud using Terraform!
+nav_order: 10
+parent: Examples
+---
+
+# Introduction
+This example will go through how to spawn a virtual machine using Terraform with Direktiv. We will use a publicly accessible [git](https://github.com/vorteil/terraform-examples). Then proceed to message a Discord webhook to let everyone know the instance is alive.
+
+```yaml
+id: spawn-instance
+functions:
+- id: git
+  image: vorteil/git:v1
+- id: discordmsg
+  image: vorteil/discordmsg:v2
+- id: tfrun
+  image: vorteil/terraform:v1
+  files:
+  - key: terraform-examples
+    scope: instance
+    type: tar.gz
+description: Clones a repository and uses terraform to deploy an instance
+states:
+  # continued in next code block
+```
+
+**NOTE: The files attribute is empty and will be filled in via the git state.**
```

Comment:
**NOTE:** The `files` attribute is empty and will be populated using the git state.